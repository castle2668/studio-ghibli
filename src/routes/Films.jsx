import React, { useEffect, useState } from "react";
import axios from "axios";
import FilmList from "@/components/FilmList/FilmList";

const App = () => {
  const [films, setFilms] = useState([]);

  const getFilms = async () => {
    try {
      const result = await axios.get("/db/data.json");
      setFilms(
        result.data.films.map((film) => {
          return {
            id: film.id,
            title: film.title,
            original_title: film.original_title,
            description: film.description,
            director: film.director,
            producer: film.producer,
            movie_banner: film.movie_banner,
            image: film.image,
            running_time: film.running_time,
          };
        })
      );
    } catch (error) {
      console.log(error);
    }
  };

  useEffect(() => {
    getFilms();
  }, []);

  return (
    <div>
      <FilmList films={films}></FilmList>
    </div>
  );
};

export default App;
