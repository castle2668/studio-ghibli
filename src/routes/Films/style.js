import styled from "styled-components";

export const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  grid-auto-rows: 1fr;
  gap: 16px;
`;
