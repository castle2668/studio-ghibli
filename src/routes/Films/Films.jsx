import React from "react";
import { useLoaderData } from "react-router-dom";

import FilmCard from "@/components/FilmCard/FilmCard";
import { getFilms } from "@/utils/films";

import * as Style from "./style";

export const loader = async () => {
  const films = await getFilms();
  return { films };
};

const Films = () => {
  const { films } = useLoaderData();

  return (
    <Style.Container>
      {(films || []).map((film) => (
        <FilmCard key={film.id} film={film} />
      ))}
    </Style.Container>
  );
};

export default Films;
