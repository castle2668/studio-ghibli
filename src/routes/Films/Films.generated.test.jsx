import React from "react";
import renderer from "react-test-renderer";

import Films, { loader } from "./Films";

jest.mock("react-router-dom");
jest.mock("@/components/FilmCard/FilmCard");
jest.mock("@/utils/films");
jest.mock("./style");

const renderTree = (tree) => renderer.create(tree);
describe("<Films>", () => {
  it("should render component", () => {
    expect(renderTree(<Films />).toJSON()).toMatchSnapshot();
  });
});

describe("loader", () => {
  it("should expose a function", () => {
    expect(loader).toBeDefined();
  });

  it("loader should return expected output", async () => {
    const retValue = await loader();
    expect(retValue).toBeTruthy();
  });
});
