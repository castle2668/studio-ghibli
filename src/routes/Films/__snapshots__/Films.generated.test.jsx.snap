// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmCard> should render component 1`] = `null`;

exports[`<Films> should render component 1`] = `null`;
